pytorch_toolkit/formula_recognition/im2latex/models/text_recognition_heads/attention_based.py

```diff
@@ -0,0 +1,280 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import math
+from collections import namedtuple, OrderedDict
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn import init
+
+from ...data.vocab import END_TOKEN, START_TOKEN
+
+INIT = 1e-2
+Candidate = namedtuple('candidate', 'score, dec_state_h, dec_state_c, output, targets, logits')
+
+
+class TextRecognitionHead(nn.Module):
+    def __init__(self, out_size, configuration):
+        super(TextRecognitionHead, self).__init__()
+        self.emb_size = configuration.get("emb_size")
+        self.enc_rnn_h = configuration.get("enc_rnn_h")
+        self.dec_rnn_h = configuration.get("dec_rnn_h")
+        self.max_len = configuration.get("max_len")
+        self.n_layer = configuration.get("n_layer")
+        self.beam_width = configuration.get('beam_width')
+        self.in_lstm_ch = configuration.get('in_lstm_ch')
+        self.rnn_encoder = nn.LSTM(self.in_lstm_ch, self.enc_rnn_h,
```

Comment:
hard to read. why not something like `encode_input_size` and `encoder_hidden_size` instead of `in_lstm_ch` and `enc_rnn_h`?

Reply:
Renamed these variables